gateway/src/main/java/io/pravega/example/iot/gateway/DataHandler.java

```diff
@@ -83,4 +93,41 @@ public String postData(@Context Request request, String data, @PathParam("remote
             throw e;
         }
     }
+
+    @POST
+    @Path("/rawData/{remoteAddr}")
+    public String postRawData(@Context Request request, byte[] data, @PathParam("remoteAddr") String remoteAddr) throws Exception {
+        final long ingestTimestamp = System.currentTimeMillis();
+        final String ingestTimestampStr = dateFormat.format(new Date(ingestTimestamp));
+        try {
+            if(data == null)
+                return "{DATA Not Received}";
+
+            // Get or calculate the routing key.
+            final String routingKeyAttributeName = Parameters.getRoutingKeyAttributeName();
+            final String routingKey;
+            if (routingKeyAttributeName.isEmpty()) {
+                routingKey = Double.toString(Math.random());
+            }else {
+                routingKey = "default_routingKey";
+            }
+
+            String message = new String(data);
```

Comment:
Are you guaranteed that new `String(data)` will always succeed? What if data contains binary data such as an Avro event? I recommend not trying to parse data in any way. Change your logging method to only log the size of the data, not the contents.

Reply:
changed to print length of the message.